src/monodroid/jni/debug.hh

```diff
@@ -9,6 +9,14 @@
 #ifdef __cplusplus
 namespace xamarin::android
 {
+	enum class DebuggerResult : unsigned int
```

Comment:
This isn't a "result" from a debugger (which would be...what? result of "breakpoint reached"?  result of "read this address"?).

This is the "status" of *connecting* the debugger.

I think `DebuggerConnectionStatus` might make more sense in-context.

Furthermore, considering that we have separate "domains" of errors and "success" codes -- granted only one error so far, but that's not necessarily going to remain true going forward -- we might want to make the "domains" more explicit:

  * Error domain: negative values
  * Unconnected, not an error: 0
  * Connected: positive values.

This in turn means changing the underlying type to a signed type.

Thus:

```cpp
enum class DebuggerConnectionStatus : int
{
    Connected = 1,
    Unconnected = 0,
    Error = -1,
    Error_CouldNotCreateThread = -2,
};
```

...though perhaps `Error_CouldNotCreateThread` is a bridge too far?

Assuming we go with that, we could write a function to wrap error checking:

```cpp
inline bool is_error (DebuggerConnectionStatus status)
{
    return status < 0; // ...assuming this can even compile?
}
```

...might be a bridge too far.

Even if we stick with { Error, Unconnected, Connected }, I think -1, 0, and 1 are "saner" values than 0, 1, 2.

Reply:
I think if there was more than one `Error` case it might be worth making the multiple enum values & helper method.

Since, it's just the once case I went with renaming everything, which I think looks good. 👍